rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/PodDeletionCostAnnotator.scala

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2017-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.rollingupdate.kubernetes
+
+import akka.actor.Actor
+import akka.actor.ActorLogging
+import akka.annotation.InternalApi
+import akka.cluster.Cluster
+import akka.cluster.ClusterEvent
+import akka.cluster.Member
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.ConnectionContext
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.HttpsConnectionContext
+import akka.pki.kubernetes.PemManagersProvider
+
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.security.KeyStore
+import java.security.SecureRandom
+import javax.net.ssl.KeyManager
+import javax.net.ssl.KeyManagerFactory
+import javax.net.ssl.SSLContext
+import javax.net.ssl.TrustManager
+import scala.collection.immutable
+import scala.collection.immutable.SortedSet
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.Await
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+@InternalApi
+final class PodDeletionCostAnnotator extends Actor with ActorLogging {
+  private val cluster = Cluster(context.system)
+  private val k8sSettings = KubernetesSettings(context.system)
```

Comment:
An alternative would be to pass in KubernetesSettings as a constructor parameter. Then no need to make KubernetesSettings an extension since it's anyway only created once by the other extension.